internal/controllers/topology/cluster/desired_state.go

```diff
@@ -75,6 +76,19 @@ func (r *Reconciler) computeDesiredState(ctx context.Context, s *scope.Scope) (*
 	}
 	s.UpgradeTracker.MachineDeployments.MarkUpgrading(mdUpgradingNames...)
 
+	if len(s.Current.MachinePools) > 0 {
```

Comment:
Nit: either adjust the above comment or add here:

```
	// Mark all the MachinePools that are currently upgrading.
	// This captured information is used for:
	// - Building the TopologyReconciled condition.
	// - Make upgrade decisions on the control plane.
	// - Making upgrade decisions on machine pools.
```

Reply:
Let's add your suggested comment here. Increases the chances that we keep it up-to-date